Update the book and examples to Twilight 0.11

Erk- opened the pull request:
This pr updates all the examples to version 0.10, and adds a small
section about `Id`.

Erk- committed:
update the book and examples to twilight 0.10

7596ff commented on src/chapter_1_crates/section_1_model.md:
```diff
@@ -25,6 +25,15 @@ and returned by the gateway API. `guild` contains types owned by the Guild
+have. This helps ensuring that IDs are not used in the wrong context. If you
+need to use the id in a different context you can use the `cast` method which
+allow changing the marker type. This is for example helpful to turn a guild ID
+into a role ID to get the `@everyone` role.
```
maybe we could add some examples of creating and casting ids?

Erk- committed:
Merge remote-tracking branch 'origin/docs' into erk/update-to-0-10

Erk- committed:
Updated to 0.11, and various cleanups.

- added example of id cast
- deprecated command parser
- moved embed builder to use util

7596ff commented on src/SUMMARY.md:
```diff
@@ -8,7 +8,6 @@
-    - [Command Parser](./chapter_1_crates/section_5_command_parser.md)
     - [Standby](./chapter_1_crates/section_6_standby.md)
     - [First-party](./chapter_1_crates/section_7_first_party/summary.md)
         - [Embed Builder](./chapter_1_crates/section_7_first_party/section_1_embed_builder.md)
```
```suggestion
```

7596ff commented on src/chapter_1_crates/section_7_first_party/section_4_util.md:
```diff
@@ -11,10 +11,12 @@ enabled via feature flags.
 
-The `builder` feature enables builders for large structs, at the
-moment only a builder for commands is availible.
+The `builder` feature enables builders for large structs. There are as
+of writing builders for [`Command`], [`Embed`] and
+[`InteractionResponseData`].
```
I think it's better to link to the builders themselves:
```suggestion
The `builder` feature enables builders for large structs. At the time of
writing, it contains the following builders:
- [`CommandBuilder`]
- [`EmbedBuilder`]
- [`InteractionResponseData`]
```

7596ff commented on src/chapter_1_crates/section_7_first_party/section_4_util.md:
```diff
@@ -106,3 +141,8 @@ let timestamp = user.timestamp();
 
 *crates.io*: <https://crates.io/crates/twilight-util>
+
+
+[`Command`]: https://api.twilight.rs/twilight_model/application/command/struct.Command.html
+[`Embed`]: https://api.twilight.rs/twilight_model/channel/embed/struct.Embed.html
+[`InteractionResponseData`]: https://api.twilight.rs/twilight_model/http/interaction/struct.InteractionResponseData.html
```
```suggestion
[`CommandBuilder`]: https://api.twilight.rs/twilight_util/builder/command/struct.CommandBuilder.html
[`EmbedBuilder`]: https://api.twilight.rs/twilight_util/builder/embed/struct.EmbedBuilder.html
[`InteractionResponseDataBuilder`]: https://api.twilight.rs/twilight_util/builder/struct.InteractionResponseDataBuilder.html
```

Erk- committed:
Apply suggestions from code review

Co-authored-by: Cassandra McCarthy <cassie@7596ff.com>

Erk- committed:
Update src/SUMMARY.md

Co-authored-by: Cassandra McCarthy <cassie@7596ff.com>

7596ff committed:
nit
